code/modules/wiremod/components/sensors/view_sensor.dm

```diff
@@ -0,0 +1,63 @@
+/**
+ * # View Sensor
+ *
+ * Returns all movable objects in view.
+ */
+
+#define VIEW_SENSOR_RANGE 5
+#define VIEW_SENSOR_COOLDOWN 1 SECONDS
+
+/obj/item/circuit_component/view_sensor
+	display_name = "View Sensor"
+	desc = "Outputs a list with all movable objects in it's view. Requires a shell."
+	category = "Sensor"
+
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL
+
+	power_usage_per_input = 10 //Normal components have 1
+
+	/// The result from the output
+	var/datum/port/output/result
+	var/datum/port/output/cooldown
+
+	COOLDOWN_DECLARE(use_cooldown)
+
+	var/see_invisible = SEE_INVISIBLE_LIVING
+
+/obj/item/circuit_component/view_sensor/populate_ports()
+	result = add_output_port("Result", PORT_TYPE_LIST(PORT_TYPE_ATOM))
+	cooldown = add_output_port("Scan On Cooldown", PORT_TYPE_SIGNAL)
+
+/obj/item/circuit_component/view_sensor/input_received(datum/port/input/port)
+	if(!COOLDOWN_FINISHED(src, use_cooldown))
```

Comment:
this cooldown needs to be circuit based, or else you can just put multiple of the same components in the same circuit and bypass cooldown